runtime/src/accounts_db.rs

```diff
@@ -1633,8 +1637,9 @@ impl AccountsDb {
         accounts_scan.stop();
 
         let mut clean_old_rooted = Measure::start("clean_old_roots");
+        let purges = [purges_in_root, purges_unrooted].concat();
```

Comment:
I can see how differentiating `purges_in_root` and `purges_unrooted` would be more self documenting, but this does eat an extra allocation. A potential alternative is to just use `purges` but rely on comments in the various places `purges` is added to for documentation

Reply:
Which would you prefer?

Another options is to call `clean_old_rooted_accounts()` twice, once for `purges_in_root` and once for `purges_unrooted`.

Reply:
or, some `iter::chain()`?

Reply:
> or, some iter::chain()?
Considered this as well, but it needs to be able to support par_iter() in `clean_old_rooted_accounts`

> I've collapsed these two vecs down into a single vec called purges_old_accounts now
this makes the most sense to me since we're handing this off to `par_iter()`